### DISCUSSION TITLE
magick separate LAB color space - exported A channel pitch black

### DISCUSSION THREAD

**Discussion by fred888651** [ImageMagick/ImageMagick#2501] [Help]
**VERSIONS:
ImageMagick-7.0.9-Q16
Windows 10 all updates
(PowerShell 5.1)** 

**_ISSUE:_**

**A channel exports a JPG file - but its all black. No details in it.**
_(L and B works fine)_

_magick.exe original.jpg -colorspace LAB -channel L -separate L_channel.JPG_   
_magick.exe original.jpg -colorspace LAB -channel A -separate A_channel.JPG_  
_magick.exe original.jpg -colorspace LAB -channel B -separate B_channel.JPG_  

(Original input file is sRGB color profile)

![image](https://user-images.githubusercontent.com/70697839/92109418-45a7a900-ede9-11ea-9583-7ced178296c0.png)

OUTPUT:

![image](https://user-images.githubusercontent.com/70697839/92108536-da110c00-ede7-11ea-9b38-3a17235fbbc8.png)

Link to original input file and output files:
https://cloud.comhem.se/s/132c6b34ea232be4fdea0963c92b575945c

**Comment by snibgo** [comment]
In "-channel A", the "A" stands for alpha, not "a*".

Personally, I use "-channel R" and "-channel G" and "-channel B" and "-channel A" (for "alpha") but no other abbreviations because they are confusing. Instead, I use "-channel 0", "-channel 1" etc.

EDIT to add: in "-channel B", the "B" stand for Blue, both "b*". But Blue is the third channel, as is b*, so it happens to give the correct result.

**Reply by fred888651** [reply]
Ahh ok! Confusing with these abbreviations...

Im gonna try to use R,G and B instead
(Also 0,1 and 2)

Well see what happens! Thanks!

**Reply by fred888651** [reply]
Seemed to solve the issue!

0 = L  1 = A  2 = B

magick.exe original.jpg -colorspace LAB -channel 0 -separate L_channel.JPG
magick.exe original.jpg -colorspace LAB -channel 1 -separate A_channel.JPG
magick.exe original.jpg -colorspace LAB -channel 2 -separate B_channel.JPG

Looks ok to me
Compared to all three channels in LAB using Photoshop and compared to output from above commands. Looks great.

Now I just need to figure out how to join these three files and put them together again :)

(Gonna be a PowerShell script that sharpens the L channel in a folder w 1 or more JPGs)
